### ISSUE TITLE
GCP DNS documentation need updating and gcp dns needs to be added to GCP example

### ISSUE THREAD

**Issue by verdurin** [ComputeCanada/magic_castle#37]
Looking at the `GCP`-specific DNS documentation at https://github.com/ComputeCanada/magic_castle/blob/master/README.md#google-cloud
in step 5 it says `In main.tf's dns module, configure the variables project_name and zone_name with their respective values as defined by your Google Cloud project.`.

However, in my copy of `main.tf` from the `5.7` `GCP` release, there are no such variables in the `dns` module.

**Comment by cmd-ntrf** [comment]
The Google Cloud DNS module is missing from the GCP example, and probably from some other examples I think. We will need to fix that, and document how to use the DNS module when it is missing.

In the meantime, you can add the following to your main.tf:

```
module "dns" {
  source           = "./dns/gcloud"
  email            = "you@example.com"
  project          = "your-project-name"
  zone_name        = "you-zone-name"
  name             = module.gcp.cluster_name
  domain           = module.gcp.domain
  public_ip        = module.gcp.ip
  rsa_public_key   = module.gcp.rsa_public_key
  sudoer_username  = module.gcp.sudoer_username
}
```

**Comment by verdurin** [comment]
Thanks for the super quick response. Here's the output I see when I try `terraform init` after adding what you suggested:
```
terraform init
Initializing modules...
- dns in dns/gcloud
- dns.acme in dns/acme
- dns.record_generator in dns/record_generator

Error: Attribute redefined

On dns/gcloud/main.tf line 3: The argument "version" was already set at
dns/gcloud/main.tf:2,3-10. Each argument may be set only once.


Error: Attribute redefined

On dns/gcloud/main.tf line 3: The argument "version" was already set at
dns/gcloud/main.tf:2,3-10. Each argument may be set only once.


Error: Attribute redefined

On dns/gcloud/main.tf line 3: The argument "version" was already set at
dns/gcloud/main.tf:2,3-10. Each argument may be set only once.


Error: Attribute redefined

On dns/gcloud/main.tf line 3: The argument "version" was already set at
dns/gcloud/main.tf:2,3-10. Each argument may be set only once.


Error: Attribute redefined

On dns/gcloud/main.tf line 3: The argument "version" was already set at
dns/gcloud/main.tf:2,3-10. Each argument may be set only once.
```

**Comment by verdurin** [comment]
Relevant contents of that file:
```
provider "google" {
  version = "~> 2.20"
  version = "< 3.0.0"
}
```

**Comment by verdurin** [comment]
```
terraform version
Terraform v0.12.20
+ provider.external v1.2.0
+ provider.google v2.20.1
+ provider.http v1.1.1
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.template v2.1.2
+ provider.tls v2.1.1
```

**Comment by cmd-ntrf** [comment]
It appears you found a bug in the release script. 

Removing the line `version = "< 3.0.0"` in  `dns/gcloud/main.tf` will fix the issue. 

When we build the Magic Castle release, the release script hardcodes the latest version available version of the provider at the time of the release. This avoids release being broken by new version of a provider. However, there is a problem when the provider already has a version defined, the script should not have appended a new version, it should have replaced it. I will file a new issue for that and work on a fix.

Thanks!

**Comment by verdurin** [comment]
Yes, I have already found that commenting out that line fixes the error.

**Comment by cmd-ntrf** [comment]
Commits 98fc41d66c969ef186f5060860acdc446df75524 and 399c3e034fa1f5aeb87df89d0f71d2b35ef7e250 fix this issue.
